### ISSUE TITLE
unable to create virtual user

### ISSUE THREAD

**Issue by fmagnan** [robgmills/ansible-pure-ftpd#1]
Hello,

many thanks for your role, it's very useful. However, I've got an error and I don't understand it. I'm new to Ansible and it's a little bit difficult for me.

I'm trying to launch playbook with these data:

```
pure_ftpd_vusers:
      - { name: "myname", password: "mypass", dir: "/var/www/mydir" }
```

And when I run this command:

```
ansible-playbook -u root -i hosts playbook.yml
```

I got this error:

```
TASK [community/debian-pure-ftpd : Check if virtual FTP user myuser exists] ***
fatal: [mytest]: FAILED! => {"changed": true, "cmd": ["pure-pw", "show", "myuser"], "delta": "0:00:00.011986", "end": "2016-09-16 11:42:35.379905", "failed": true, "rc": 16, "start": "2016-09-16 11:42:35.367919", "stderr": "Unable to fetch info about user [myuser] in file [/etc/pure-ftpd/pureftpd.passwd]", "stdout": "", "stdout_lines": [], "warnings": []}
```

Can you tell me what is going please?

**Comment by bjornoss** [comment]
Hi, 
I had the same problem as you.
I'm using Ansible 2.1 and I think the problem is related to Ansible 2.x. 
The problem was fixed by adding `ignore_errors: true` to the `Check if virtual FTP user exists` task:

```
- name: Check if virtual FTP user {{ item.name }} exists
  command: pure-pw show {{ item.name }}
  register: r_userexists
  when: r_passwd.stat.exists
  changed_when: "r_userexists.rc != 0"
  ignore_errors: true
```

**Comment by fmagnan** [comment]
Hi Bjornoss,
many thanks for your reply!
I confirm what you said. I'm using Ansible 2.1.1.0 on Ubuntu 16.04 and it works when I follow your instructions.

Do you know if we need to make a pull request for this?

**Comment by bjornoss** [comment]
Good to hear it worked for you as well. 
Sure, we could make a pull request, I do not know what @robgmills prefer? 
It is a very small change so maybe he just changes it without a pull request. 
/b

**Comment by robgmills** [comment]
Sorry for the delay!  I must have lost the issue in my inbox.  Feel free to make a PR.  Happy to include it.

**Comment by robgmills** [comment]
Ah, scratch that.  I'll make the change.  One second.

**Comment by robgmills** [comment]
So I suspect that this is an incompatibility between Ansible 2.x, the version of python and tools installed from the Xenial apt repository, and the version of python and tools expected by python packages installed by pip (in this case `pexpect`).  This is the third time I've encountered it when dealing with the 16.xx versions of Ubuntu vs Debian Jessie.  I'm not saying that it's a bad release from Canonical, but there's a big jump in python versions in the Xenial apt repositories that appears to be the only relevant difference.

I've incorporated the suggested fix for now and tagged it as version `1.0.1`.  But I'll likely remove it in favor of another fix later.

[Travis-CI build](https://travis-ci.org/robgmills/ansible-pure-ftpd/builds/164649650)

Thanks for raising the issue and contributions!
